moralization/model_manager.py

```diff
@@ -0,0 +1,88 @@
+from abc import ABC, abstractmethod
+from pathlib import Path
+from typing import Union, Optional, Dict, Any
+from moralization.data_manager import DataManager
+import os
+import huggingface_hub
+
+
+class ModelManager(ABC):
+    """
+    Create, import, modify, train and publish a model.
+
+    Models can be trained on data from a DataManager, and published to hugging face.
+    """
+
+    @abstractmethod
+    def __init__(self, model_path: Union[str, Path]):
+        self._model_path = Path(model_path)
+
+    @property
+    def model_path(self) -> Path:
+        """
+        The folder where the model is stored
+        """
+        return self._model_path
+
+    def __repr__(self) -> str:
+        return f"ModelManager('{self.model_path}')"
+
+    @abstractmethod
+    def train(self, data_manager: DataManager) -> None:
+        """Train the model on the data contained in `data_manager`.
+
+        Args:
+            data_manager (DataManager): the DataManager that contains the training data
+        """
+        raise NotImplementedError
```

Comment:
To me it looked like you would either use `abc` and the `@abstractmethod` decorator, or raise a `NotImplementedError`, but not both. Just stumbled about it as it seems to duplicate functionality and also decreases test coverage.